### PR TITLE
Possible to trigger TOTP activation and login using TOTP

### PR DESCRIPTION
**Pull request by alneberg**
Before submitting a pr:
- [x] Tests passing
- [x] Black formatting
- [x] Rebase/merge the `dev` branch
- [x] Note in the CHANGELOG

### CODE REVIEW

**Review by inaod568 — CHANGES_REQUESTED** [review]

File: dds_cli/user.py [hunk 1]
```diff
@@ -138,42 +139,95 @@ def __authenticate_user(self):
...
+                try:
+                    response = requests.get(
+                        dds_cli.DDSEndpoint.SECOND_FACTOR,
+                        headers={"Authorization": f"Bearer {partial_auth_token}"},
+                        json=json_request,
+                        timeout=dds_cli.DDSEndpoint.TIMEOUT,
+                    )
+                    response_json = response.json()
+                except requests.exceptions.RequestException as err:
+                    raise exceptions.ApiRequestError(message=str(err)) from err
```
[the remark this answers is not in the document]
Comment (aanil) [reply dds_cli/user.py:222]: There's an extra try again step here thats not in the `request_get()` func
`try_again = rich.prompt.Confirm.ask(
                            "Second factor authentication failed, would you like to try again?"`

**Review by inaod568 — COMMENTED** [review]
Seems to be a bug somewhere. The one-time code is never accepted.
```
root@a119c8420857:/code# dds auth twofactor
     ︵ 
 ︵ (  )   ︵ 
(  ) ) (  (  )   SciLifeLab Data Delivery System 
 ︶  (  ) ) (    http://dds_backend:5000/ 
      ︶ (  )    Version 1.0.3 
          ︶
INFO     Starting configuration of one-time authentication code method.                                         
? Which method would you like to use? Authenticator App
INFO     No saved token found, or token has expired, proceeding with authentication                             
DDS username: unituser_1
DDS password: 
INFO     Please enter the one-time authentication code sent to your email address (leave empty to exit):        
Authentication one-time code: 14367889
Authentication one-time code: 14367889
Authentication one-time code: 
```

**Comment by inaod568** [comment]
@alneberg @aanil Also, all requests now go through the `perform_request` function in the `utils.py` module

**Comment by alneberg** [comment]
Yes, the bug seems to be due to an indentation misstake in the merge with dev. Should be easy to fix.

**Comment by inaod568** [comment]
@alneberg @aanil You can add the following to a `codecov.yml` file in the repo root, this is the same as we have in the `dds_web`
```
coverage:
  status:
    project:
      default:
        threshold: 5%
    patch: off
```

**Comment by inaod568** [comment]
@zishanmirza commented the following in the Slack `dds-development` channel:

> I noticed the cli doesn't have tests for TOTP. Would be great to add that to the PR.

**Review by inaod568 — CHANGES_REQUESTED** [review]
Works well! If you could add a couple of tests though, that would be great! You don't have to add heaps, but look at the existing tests regarding `perform_request` and a couple new ones for this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -146,14 +146,18 @@ Please add a _short_ line describing the PR you make, if the PR implements a spe
 
 - Patch: Add a message when the project access would be fixed for a user. ([#446](https://github.com/ScilifelabDataCentre/dds_cli/pull/446))
 
-
 ## Sprint (2022-04-06 - 2022-04-20)
 
 - `motd` command to add new message of the day via new endpoint ([#449](https://github.com/ScilifelabDataCentre/dds_cli/pull/449))
 - Patch: Message in docstrings to urge users to reauthenticate before upload and download ([#450](https://github.com/ScilifelabDataCentre/dds_cli/pull/450))
-- Pin versions in `requirements-dev.txt`: New version of `sphinx-click` makes `:nested: full` not work anymore (direct commit: https://github.com/ScilifelabDataCentre/dds_cli/commit/b91332b43e9cdee40a8132eab15e2fea3201bab6)
+- Pin versions in `requirements-dev.txt`: New version of `sphinx-click` makes `:nested: full` not work anymore (direct commit: <https://github.com/ScilifelabDataCentre/dds_cli/commit/b91332b43e9cdee40a8132eab15e2fea3201bab6>)
```
Comment (inaod568) [inline CHANGELOG.md:153]:
```suggestion
- Pin versions in `requirements-dev.txt`: New version of `sphinx-click` makes `:nested: full` not work anymore (direct commit: https://github.com/ScilifelabDataCentre/dds_cli/commit/b91332b43e9cdee40a8132eab15e2fea3201bab6)
```

**Review by inaod568 — APPROVED** [review]
Looks great here too!

**Pull request merged**